operation/manager.go

```diff
@@ -139,21 +153,10 @@ func (sm *SingleOperationManager) WaitForSuccess(
 	}
 }
 
-func (sm *SingleOperationManager) cancelInLock(ctx context.Context) {
-	myOp := ctx.Value(somCtxKeySingleOp)
-	op := sm.currentOp
-
-	if op == nil || myOp == op {
-		return
-	}
-
-	op.cancelFunc()
-
-	sm.currentOp = nil
-}
-
 type anOp struct {
 	ctx        context.Context
 	cancelFunc context.CancelFunc
 	closed     bool
+	// Used with `SingleOperationManager.mu`.
+	closedCond *sync.Cond
```

Comment:
[q] why a pointer to a `sync.Cond`? Is it also necessary for inclusion in anOP rather than singelOperationManager?